flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkTableSource.java

```diff
@@ -422,8 +422,9 @@ public void testFilterPushDownInNull() {
     Assert.assertEquals("Should have 1 record", 1, result.size());
     Assert.assertEquals(
         "Should produce the expected record", Row.of(1, "iceberg", 10.0), result.get(0));
+    String expectedScan = "ref(name=\"data\") == \"iceberg\"";
```

Comment:
Currently Flink 1.17 can push this down.

Reply:
In previous versions, Flink did not push down the `IN ()` to the source table（ `@Override
  public Result applyFilters(List<ResolvedExpression> flinkFilters) `) if it contained `NULL`.  This time it removes the `NULL` and pushes the rest down.

I'm not sure whether this change was done by Flink itself or because it upgraded the version of Calcite to to 1.29.0. Let me look into it.

Reply:
> you are saying Flink skips the `null` for predicate pushdown? if yes, please add it the comment. it will also be great we can link to a Flink jira (if there is) for future improvement on pushdown with `null` value.

As a side note, NULL has no meaning in `in ()`, and it cannot query data which is  NULL, it is correct to ignore it here. So we don't have to push it down.

Reply:
I am not saying the behavior is wrong. maybe just add a comment in the code to explain it a bit. E.g. your note above is perfect. In SQL, null check can only be done as `IS NULL` or `IS NOT NULL`.

Reply:
Sorry for the misunderstanding. Comments have been added.